### PR TITLE
Add board documentation [WIP]

### LINKED ISSUE #3347: Add documentation on test boards
### Test Board Documentation

Add notes to the current test setup that documents the test board wiring (which is -- I think -- a strict superset of what is there already). Also, I'll commit the schematic. I can include the manufacturing files, but I don't think that it is worth it for the first revision.

### PR DESCRIPTION
**Pull request by pjsg**
Fixes #3347

Make sure all boxes are checked (add x inside the brackets) when you submit your contribution, remove this sentence before doing so.

- [x] This PR is for the `dev` branch rather than for the `release` branch.
- [x] This PR is compliant with the [other contributing guidelines](https://github.com/nodemcu/nodemcu-firmware/blob/dev/CONTRIBUTING.md) as well (if not, please describe why).
- [x] The code changes are reflected in the documentation at `docs/*`.

This is adding documentation on the test board... This is a work in progress....

### CODE REVIEW

**Comment by marcelstoer** [comment]
There is some overlap in terms of documentation with @nwf's work in #3353.

**Comment by HHHartmann** [comment]
Resolved the conflicts with nwf s PR in tests/README.md . Hope I didn't mess things up.
For the rest: LGTM

**Comment by nwf** [comment]
Seems fine.

File: tests/HardwareTestHarness.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
+Hardware Test Harness
+=====================
+
+There is an implementation of the hardware testing design which is a small 4in x 4in board with positions for
+two Wemos D! Mini ESP8266 boards, a breadboard area and a number of positions for peripherals. The schematics
+are in [schematic](Test-harness-schematic-v1.pdf) and a rendering of the board is ![Board Render](Test-Harness-Render-V1.png).
```
Comment (marcelstoer) [inline tests/HardwareTestHarness.md:6]: Just realized that `Test-Harness-Render-V1.png` is missing but I vaguely remember @pjsg posted an image in a comment to some test-related issue. I failed to dig that one out, though.

**Comment by pjsg** [comment]
I'll dig it out -- I thought that it was part of the PR, but I probably
forgot the `git add`

File: tests/HardwareTestHarness.md (hunk 1 above)
Comment (marcelstoer) [reply tests/HardwareTestHarness.md:6]: Found it: https://github.com/nodemcu/nodemcu-firmware/pull/3324#issuecomment-735282277. I'll add it to `dev` now.

**Pull request merged**